File: explorer/src/main/java/com/duongame/adapter/ComicPagerAdapter.java
package com.duongame.adapter;

import android.app.Activity;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.duongame.R;
import com.duongame.fragment.BaseFragment;
import com.duongame.fragment.ExplorerFragment;
import com.duongame.fragment.HistoryFragment;
import com.duongame.fragment.SearchFragment;

import java.util.HashMap;

/**
 * Created by namjungsoo on 2016. 12. 30..
 */

public class ComicPagerAdapter extends FragmentPagerAdapter {
    private final static boolean DEBUG = false;
    private final static String TAG = "ComicPagerAdapter";
    private final int PAGE_COUNT = 3;

    private Activity context;
    private HashMap<Integer, BaseFragment> fragmentMap = new HashMap<>();
    private int lastPosition = -1;

    public ComicPagerAdapter(FragmentManager fm, Activity context) {
        super(fm);
        this.context = context;
    }

    @Override
    public int getCount() {
        //TODO: 나중에 동적으로 수정해야 한다.
        return PAGE_COUNT;
    }

    @Override
    public Fragment getItem(int position) {
        BaseFragment fragment = null;
        fragment = fragmentMap.get(position);

        if (fragment == null) {
            switch (position) {
                case 0:
                    fragment = new ExplorerFragment();
                    fragmentMap.put(position, fragment);
                    break;
                case 1:
                    fragment = new HistoryFragment();
                    fragmentMap.put(position, fragment);
                    break;
                case 2:
                    fragment = new SearchFragment();
                    fragmentMap.put(position, fragment);
                    break;
            }
        }

        return fragment;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return context.getResources().getString(R.string.explorer);
            case 1:
                return context.getResources().getString(R.string.history);
            case 2:
                return context.getResources().getString(R.string.search);
        }
        return "";
    }

    @Override
    public void setPrimaryItem(ViewGroup container, int position, Object object) {
        super.setPrimaryItem(container, position, object);

        if (position != lastPosition) {
            lastPosition = position;
            if (fragmentMap.containsKey(position)) {
                final BaseFragment fragment = fragmentMap.get(position);
                fragment.onRefresh();
            }
        }
    }

//    @Override
//    public boolean isViewFromObject(View view, Object object) {
//        return view == object;
//    }
}

File: Explorer/src/main/java/com/duongame/adapter/ComicPagerAdapter.kt
package com.duongame.adapter

import android.app.Activity
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.duongame.R
import com.duongame.fragment.BaseFragment
import com.duongame.fragment.ExplorerFragment
import com.duongame.fragment.HistoryFragment
import com.duongame.fragment.SearchFragment

/**
 * Created by namjungsoo on 2016. 12. 30..
 */
class ComicPagerAdapter(fm: FragmentManager, private val context: Activity) : FragmentPagerAdapter(fm) {
    private val PAGE_COUNT = 3
    private val fragmentMap = HashMap<Int, BaseFragment>()
    private var lastPosition = -1
    override fun getCount(): Int {
        //TODO: 나중에 동적으로 수정해야 한다.
        return PAGE_COUNT
    }

    override fun getItem(position: Int): Fragment {
        var fragment: BaseFragment? = null
        fragment = fragmentMap[position]
        if (fragment == null) {
            when (position) {
                0 -> {
                    fragment = ExplorerFragment()
                    fragmentMap[position] = fragment
                }
                1 -> {
                    fragment = HistoryFragment()
                    fragmentMap[position] = fragment
                }
                2 -> {
                    fragment = SearchFragment()
                    fragmentMap[position] = fragment
                }
            }
        }
        return fragment!!
    }

    override fun getPageTitle(position: Int): CharSequence? {
        when (position) {
            0 -> return context.resources.getString(R.string.explorer)
            1 -> return context.resources.getString(R.string.history)
            2 -> return context.resources.getString(R.string.search)
        }
        return ""
    }

    override fun setPrimaryItem(container: ViewGroup, position: Int, `object`: Any) {
        super.setPrimaryItem(container, position, `object`)
        if (position != lastPosition) {
            lastPosition = position
            if (fragmentMap.containsKey(position)) {
                val fragment = fragmentMap[position]
                fragment!!.onRefresh()
            }
        }
    }
    //    @Override
    //    public boolean isViewFromObject(View view, Object object) {
    //        return view == object;
    //    }
    companion object {
        private const val DEBUG = false
        private const val TAG = "ComicPagerAdapter"
    }
}